---
title: "ConsensusPeak"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{ConsensusPeak}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Overview 

Uncovering the locations where proteins bind to the genome is essential for 
elucidating the regulatory mechanisms that control gene expression and influence 
cellular function, development, and disease. Techniques such as ChIP-seq have 
revolutionised our ability to precisely map these interactions. However, as
single experiments are subject to substantial biological and technical
variability, it is considered best practice to generate multiple biological 
replicates.

Given multiple biological replicates, the goal is to derive a single, 
high-confidence set of peaks representing the 'true' locations that your
protein binds. While several tools have been developed towards this goal, they 
are dispersed across various packages and require diverse inputs, presenting a 
challenge in terms of ease-of-use. To address this issue, we 
introduce *ConsensusPeak*, an R package that aggregates several consensus peak 
calling metrics within a cohesive and straightforward interface.

<!-- \begin{itemize} -->
<!--   \item Irreproducible Discovery Rate (IDR) (n = 2) -->
<!--   \begin{enumerate} -->
<!--     \item Optimal -->
<!--     \item Conservative -->
<!--   \end{enumerate} -->
<!--   \item Multiple Sample Peak Calling (MSPC)  -->
<!--   \item Overlapping peak calls (n < 5) -->
<!-- \end{itemize} -->

```{r include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
# ConsensusPeak can currently only be installed from GitHub
#if(!require("remotes")) install.packages("remotes")
#remotes::install_github("neurogenomics/ConsensusPeak")

library(ConsensusPeak)
```

The package is distributed with small example data sets. These can be loaded 
with the following commands:
```{r eval=TRUE}
input1 <- system.file("extdata", "r1_creb_chr22.bam", package = "ConsensusPeak")
input2 <- system.file("extdata", "r2_creb_chr22.bam", package = "ConsensusPeak")
input3 <- system.file("extdata", "r3_creb_chr22.bam", package = "ConsensusPeak")
```

A popular way to generate consensus peaks is via the Irreproducible Discovery 
Rate (IDR). This method is a measure of peak rank consistency between 
two replciates, and is used in the ENCODE project. Here, we implement two 
variants of IDR: conservative and optimal. Despite its popularity, IDR analysis 
can only be performed with two replicates.
```{r eval=TRUE}

idr_analysis(treat_files = c(input1, input2), # BAM files
             control_files = NULL,
             type = "all", # all runs both the conservative and optimal methods. 
             is_paired = FALSE,
             out_dir = ".",
             nomodel = TRUE
             )
```

We have implemented two methods that can handle more than two replicates. These 
are MSPC and ChIP-R. Note that MSPC requires .NET 6.0 or higher to be installed 
on your system.

```{r eval=TRUE}

multiple_replicates_mspc(treat_files = c(input1, input2, input3),
                         out_dir = ".",
                         subdir_name = "mspc_analysis",
                         is_paired = FALSE,
                         replicateType = "Biological",
                         stringencyThreshold = 1e-8,
                         weakThreshold = 1e-4,
                         c = 3,
                         nomodel = TRUE
                         )
```

```{r eval=TRUE}

multiple_replicates_chipr(treat_files = c(input1, input2, input3),
                          is_paired = FALSE,
                          out_dir = ".",
                          nomodel = TRUE
                          )
```







